src/main/java/com/exactpro/th2/crawler/Crawler.java

```diff
@@ -92,102 +111,109 @@ public Crawler(@NotNull CradleStorage storage, @NotNull DataServiceService dataS
         this.configuration = Objects.requireNonNull(configuration, "Crawler configuration cannot be null");
         this.from = Instant.parse(configuration.getFrom());
         this.floatingToTime = configuration.getTo() == null;
-        this.to = floatingToTime ? Instant.now() : Instant.parse(configuration.getTo());
+        this.workAlone = configuration.getWorkAlone();
+        this.crawlerTime = new CrawlerTimeImpl();
+        this.to = floatingToTime ? crawlerTime.now() : Instant.parse(configuration.getTo());
         this.defaultIntervalLength = Duration.parse(configuration.getDefaultLength());
         this.numberOfEvents = this.numberOfMessages = 0L;
         this.sleepTime = configuration.getDelay() * 1000;
+
+        prepare();
     }
 
-    public void start() throws InterruptedException, IOException, UnexpectedDataServiceException {
-        String name = configuration.getName();
-        String type = configuration.getType();
-        int batchSize = configuration.getBatchSize();
+    private void prepare() {
+        String crawlerName = configuration.getName();
+        crawlerType = configuration.getType();
+        batchSize = configuration.getBatchSize();
 
-        if (!(EVENTS.equals(type) || MESSAGES.equals(type))) {
+        if (!(EVENTS.equals(crawlerType) || MESSAGES.equals(crawlerType))) {
             throw new ConfigurationException("Type must be either EVENTS or MESSAGES");
         }
 
         if (!floatingToTime && Duration.between(from, to).abs().compareTo(defaultIntervalLength) < 0)
             throw new IllegalArgumentException("Distance between \"from\" and \"to\" parameters cannot be less" +
                     "than default length of intervals");
 
-        CrawlerId crawlerId = CrawlerId.newBuilder().setName(name).build();
+        crawlerId = CrawlerId.newBuilder().setName(crawlerName).build();
         CrawlerInfo crawlerInfo = CrawlerInfo.newBuilder().setId(crawlerId).build();
 
         LOGGER.info("Crawler started working");
 
-        DataServiceInfo info = dataService.crawlerConnect(crawlerInfo);
-
-        while (!Thread.currentThread().isInterrupted()) {
-
-            String dataServiceName = info.getName();
-            String dataServiceVersion = info.getVersion();
+        info = dataService.crawlerConnect(crawlerInfo);
+    }
 
-            interval = getOrCreateInterval(dataServiceName, dataServiceVersion, type);
+    public long process() throws IOException, UnexpectedDataServiceException {
```

Comment:
I would suggest using `Duration` instead of `long` value. It is more clear is my opinion